pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go

```diff
@@ -318,9 +327,51 @@ func getIgnitionConfig(profile *performancev2.PerformanceProfile, pinningMode *a
 		Name:     getSystemdService(clearIRQBalanceBannedCPUs),
 	})
 
+	if ok, ovsSliceName := MoveOvsIntoOwnSlice(); ok {
+		// Create the OVS slice that will lift the cpu restrictions for better kernel networking performance
+		// This is technically not necessary as systemd is smart enough
+		// to create a slice when a unit references it.
+		// However, this allows us to set the resources allocated and the cpu balancing
+
+		ovsCgroupUnit, err := getOvsSliceDefinition(ovsSliceName)
+		if err != nil {
+			return nil, err
+		}
+
+		ovsMode := 0644
+		addContent(ignitionConfig, ovsCgroupUnit, "/etc/systemd/system/"+ovsSliceName, &ovsMode)
+
+		// Configure OVS services to use the newly created slice
+		serviceOvsSlice, err := getOvsSliceUsage(ovsSliceName)
+		if err != nil {
+			return nil, err
+		}
+
+		addContent(ignitionConfig, serviceOvsSlice, "/etc/systemd/system/openvswitch.service.d/"+templateOvsSliceUsageFile, &ovsMode)
+		addContent(ignitionConfig, serviceOvsSlice, "/etc/systemd/system/ovs-vswitchd.service.d/"+templateOvsSliceUsageFile, &ovsMode)
+		addContent(ignitionConfig, serviceOvsSlice, "/etc/systemd/system/ovsdb-server.service.d/"+templateOvsSliceUsageFile, &ovsMode)
+
+		// Tell OVN-K to enable dynamic cpu pinning
+		content, err := getTemplatedOvsFile(assets.Configs, filepath.Join("configs", ovsDynamicPinningTriggerFile), ovsSliceName)
+		if err != nil {
+			return nil, err
+		}
+		addContent(ignitionConfig, content, ovsDynamicPinningTriggerHostFile, &ovsMode)
+	}
+
 	return ignitionConfig, nil
 }
 
+func MoveOvsIntoOwnSlice() (bool, string) {
+	// Make sure this does not interfere with SNO and workload partitioning
```

Comment:
So restricting OVS to reserved cpus would be done at later stage by the OVN-K if it sees that WP enabled?

Reply:
It will not specifically care about WP, rather WP will make sure the affinity of the pod is right and the pod will just apply it to OVS too.

This will actually allow the accelerated startup to work properly, because the affinity will be wide during the boot sequence and then restricted once the accelerated startup is reverted.

Reply:
The ovn-k pods is passing the information about the allowed cpus to ovs?

Reply:
Yes. The ovnkube-node Pod is setting the OVS affinity.